nullaway/src/main/java/com/uber/nullaway/generics/GenericsChecks.java

```diff
@@ -423,16 +425,54 @@ private static void reportInvalidOverridingMethodParamTypeError(
    * @param analysis the analysis object
    * @param state the visitor state
    */
-  public static void checkTypeParameterNullnessForAssignability(
-      Tree tree, NullAway analysis, VisitorState state) {
+  public void checkTypeParameterNullnessForAssignability(
+      Tree tree, NullAway analysis, VisitorState state, Config config) {
     if (!analysis.getConfig().isJSpecifyMode()) {
       return;
     }
     Type lhsType = getTreeType(tree, state);
     Tree rhsTree;
+    // update inferredTypes cache for assignments
     if (tree instanceof VariableTree) {
       VariableTree varTree = (VariableTree) tree;
       rhsTree = varTree.getInitializer();
+      Tree lhsTree = varTree.getType();
+
+      if (rhsTree instanceof MethodInvocationTree) {
+        MethodInvocationTree methodInvocationTree = (MethodInvocationTree) rhsTree;
+        Symbol.MethodSymbol methodSymbol = ASTHelpers.getSymbol(methodInvocationTree);
+        // rhs is generic method call, no explicit generic arguments, lhs type has generic
+        if (methodSymbol.type instanceof Type.ForAll
+            && methodInvocationTree.getTypeArguments().isEmpty()
+            && lhsTree instanceof ParameterizedTypeTree) {
+          List<? extends Tree> lhsTypeArguments =
+              ((ParameterizedTypeTree) lhsTree).getTypeArguments();
+          // method call has a return type of class type
+          if (methodSymbol.getReturnType() instanceof Type.ClassType) {
```

Comment:
How can this condition be false?  If a type variable is returned?  But don't we want to handle that case?